Command.fs

```diff
@@ -125,56 +102,24 @@ module Compose =
 
     /// Gets the highest intermediate number for some variable in a given
     /// int expression
-    and getIntIntermediate v =
+    let rec getIntIntermediate v =
         function
         | AVar (Reg (Intermediate (n, x))) when v = x-> Some n
         | AVar (Sym { Params = xs } ) ->
             Seq.fold maxOpt None <| (Seq.map (getIntermediate v) <| xs)
-        | AVar _ | AInt _ -> None
         | AAdd xs | ASub xs | AMul xs ->
             Seq.fold maxOpt None <| (Seq.map (getIntIntermediate v) <| xs)
-        | ADiv (x, y) ->
+        | ADiv (x, y) | AMod (x, y) ->
             maxOpt (getIntIntermediate v x) (getIntIntermediate v y)
         | _ -> None
 
-    /// <summary>
-    ///     Finds the highest intermediate stage number in a Boolean expression.
-    ///     Returns one higher.
-    /// </summary>
-    /// <param name="_arg1">
-    ///     The <c>BoolExpr</c> to investigate.
-    /// </param>
-    /// <returns>
-    ///     The next available intermediate stage number.
-    ///     If the expression has no intermediate stages, we return 0.
-    /// </returns>
-    and nextBoolIntermediate : BoolExpr<Sym<_>> -> bigint =
-        function
-        | BVar (Reg (Intermediate (n, _))) -> n + 1I
-        | BVar (Sym { Params = xs } ) ->
-            xs |> Seq.map nextIntermediate |> Seq.fold (curry bigint.Max) 0I
-        | BVar _ -> 0I
-        | BAnd xs | BOr xs ->
-            xs |> Seq.map nextBoolIntermediate |> Seq.fold (curry bigint.Max) 0I
-        | BImplies (x, y) ->
-            bigint.Max (nextBoolIntermediate x, nextBoolIntermediate y)
-        | BNot x -> nextBoolIntermediate x
-        | BGt (x, y) | BLt (x, y) | BGe (x, y) | BLe (x, y) ->
-            bigint.Max (nextIntIntermediate x, nextIntIntermediate y)
-        | BEq (x, y) ->
-            bigint.Max (nextIntermediate x, nextIntermediate y)
-        | BTrue | BFalse -> 0I
-
     /// Gets the highest intermediate number for some variable in a given
     /// boolean expression
     and getBoolIntermediate v =
         function
         | BVar (Reg (Intermediate (n, name))) when name = v -> Some n
         | BVar (Sym { Params = xs } ) ->
             Seq.fold maxOpt None <| (Seq.map (getIntermediate v) <| xs)
-        | BVar (Reg (After x)) when x = v -> None
-        | BVar (Reg (Before x)) when x = v -> None
-        | BVar (Reg (Intermediate(i, x))) when x = v -> Some i
```

Comment:
You appear to have deleted `| BVar (Reg (Intermediate(i, x))) when x = v -> Some i`

Surely that's required to get the intermediate value from an actual intermediate variable?

Or am I mis-reading this ?